library/Makefile

```diff
@@ -35,9 +35,12 @@ SOEXT_TLS=so.10
 SOEXT_X509=so.0
 SOEXT_CRYPTO=so.1
 
-DLEXT=so
-# OSX shared library extension:
-# DLEXT=dylib
+# Set DLEXT=dylib to compile as a shared library for Mac OS X
+DLEXT ?= so
```

Comment:
What's the point of this change? Setting a variable on the command line overrides a setting inside the makefile, even with the `=` operator. The only difference between `=` and `?=` is what happens if there are multiple assignments inside the same makefile.

Reply:
@gilles-peskine-arm: You are correct in that you can override the value of an assignment with `=` by using command line arguments such as:
```
make -C library DLEXT=dylib
```
However, using the `=` operator overrides environment variables. Therefore, without this change the following would not work to compile using Mac OS X:
```
DLEXT=dylib make -C library
```
I suppose is a matter of preference. I personally tend to use environment variables more often than variable assignment as argument to make. Furthermore, having the `?=` clearly notes that this is one of the non-conventional variables from this Makefile that the user can configure.